Limit the heap size of postgresql containers to prevent OOM

zugao opened the pull request:
Force limit java heap in all containers of PostgreSQL instances deployed by Stackgres in order to prevent OOM and eliminate PostgreSQL instance restarts. The PR will temporarily resolve the issue from this [bug report](https://gitlab.com/ongresinc/stackgres/-/issues/2452) until Stackgres fixes it in the next version.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

<!--
Thank you for your pull request. Please provide a description above and
review the checklist.

Contributors guide: ./CONTRIBUTING.md

Remove items that do not apply. For completed items, change [ ] to [x].
These things are not required to open a PR and can be done afterwards,
while the PR is open.
-->

Commit:
Limit the heap size of postgresql containers to prevent OOM

wejdross commented:
where is that going to be passed?

Kidswiss commented on class/defaults.yml:
```diff
@@ -22,6 +22,13 @@ parameters:
+          # The value is applied to all containers of a pod in a PostgreSQL instance.
+          # This is a workaround for Stackgres v1.5.0 due to OOM issues in cluster-controller containers.
+          # Bug report - https://gitlab.com/ongresinc/stackgres/-/issues/2452
+          - -XX:MaxHeapSize=512m
```
Do we really need such a high size though? AFAIK only the `cluster-controller` is using java in the pod. Or does this also apply to the operator itself?

zugao replied:
This also applies to stackgres operator.

Update: an to everything

zugao commented:
> where is that going to be passed?

This will be passed to the Stackgres operator which in turn will apply it to itself and to all `PostgreSQL` clusters created via `SGCluster` CRD.

Kidswiss commented:
Maybe one more thing: does this setting work with the native image? Or do you need to switch back to the jvm one?

Kidswiss commented:
As discussed, I will let this run on my home cluster over the weekend with a max heap setting of 80m.

If I don't see any OOMs or restarts until monday I'd propose to go with a similar value, so that we can save on resources

zugao commented:
> Maybe one more thing: does this setting work with the native image? Or do you need to switch back to the jvm one?

As we figured out together, this will work only on native image.

wejdross commented:
@Kidswiss how was Your test?

zugao commented:
So the final test was positive. You can check the [issue](https://gitlab.com/ongresinc/stackgres/-/issues/2452#note_1587570059) on Stackgres. I removed the jvm arguments because they are useless and moved from native to jvm image. Now the cluster-controller memory is limited by the resource limit of the container.
